### ISSUE TITLE
Check config failures_before_critical invalid

### ISSUE THREAD

**Issue by jakubgs** [hashicorp/consul#7114]
I tried setting `failures_before_critical` for one of my check but I got back:
```
Error reloading: Unexpected response code: 500 (1 error occurred:
	* Failed to reload configs
```
And in the logs:
```
consul[X]: ==> Error parsing /etc/consul/service_abc.json: 1 error occurred:
consul[X]:         * invalid config key services[0].checks[0].failures_before_critical
```
Also, the documentation seems weird:
https://www.consul.io/docs/agent/checks.html
It states that the config should be added like this:
```
{
  "checks": {
    ...
    "success_before_passing" : 3
    "failures_before_critical" : 3
  },
}
```
But `checks` is an array of check objects, not an object itself.

**Comment by blake** [comment]
Hi @jakubgs,

What version of Consul are you using? These two options were recently added to Consul, and are available in the current 1.7.0 beta.

You're right that the docs incorrect. `checks` should be an array of objects. Would you like to open a PR to fix this?

**Comment by jakubgs** [comment]
>What version of Consul are you using? These two options were recently added to Consul, and are available in the current 1.7.0 beta.

You update your official docs to match beta version instead of the stable version? Weird.

>Would you like to open a PR to fix this?

Can I invoice you for it?

**Comment by blake** [comment]
> You update your official docs to match beta version instead of the stable version? Weird. 

Valid feedback. Other areas of the docs call out the minimum version required to use a particular Consul feature. We should do the same here to avoid confusion.

> > Would you like to open a PR to fix this?
> 
> Can I invoice you for it?

No worries. I can take care of this. :-)
